lib/encoder/include/AEAT6012.h

```diff
@@ -24,16 +24,18 @@ class AEAT6012 : public Encoder {
 
   AEAT6012(const Config &config);
 
-  // Trigger a blocking encoder read and retrieve the absolute position in degrees
-  // Returns whether the read was successful
-  bool getAngleDeg(float &angle) override;
+  // Trigger a blocking encoder read and store a value for angle and velocity
+  // returns whether the read was successful
+  [[nodiscard]] bool update() override;
```

Comment:
this nodiscard qualifier enforces that we cannot ignore error reports. currently, i've added this to sensors and encoders. we should add this to the actuatorcontroller classes as part of the previously mentioned task.

unfortunately, qualifiers are not inherited so this has to go in a couple places, but i think this is greatly needed to make sure we dont miss any errors